### ISSUE TITLE
Allow for a margin when comparing schedule

### ISSUE THREAD

**Issue by jonkerj** [talos-systems/talos-controller-manager#40]
When I try the controller-manager, it seems every queuing of an upgrade leads to confusion about the time:

```
[talos-controller-manager-j5b25] 2020-09-10T13:01:43.372Z       INFO    controllers.Pool        obtained version for pool       {"version": "v0.6.0-beta.3", "pool": "", "channel": "beta"}
[talos-controller-manager-j5b25] 2020-09-10T13:01:43.380Z       INFO    controllers.Pool        requeuing upgrade       {"when": "2020-09-10T13:03:43.380Z", "pool": ""}
[talos-controller-manager-j5b25] 2020-09-10T13:01:43.389Z       INFO    controllers.Pool        skipping reconciliation, next run is in the future      {"pool": ""}
```

If the timestamp in the log is accurate, the message is right: it is 0.017s earlier then scheduled. In my case, literally every schedule over a couple of hours is affected by this. It could well be that my clock is a bit flaky (these are VMs running on a tightly overprovisioned hypervisor).

Looking at [the code](https://github.com/talos-systems/talos-controller-manager/blob/1dd8566660be0a1b8f39369c5c7d24117238116b/pkg/controllers/pool_controller.go#L154-L156), it looks like it can be easily solved with a different comparison function. Maybe just adding 1 second to `Now().UTC()`.

**Comment by jonkerj** [comment]
Potentially interesting observation: the scheduled time is consistently in the future (at least for several hours, every 2min). Maybe there is matching delay somewhere in the part where the task is actually scheduled.

**Comment by Ulexus** [comment]
This does not actually appear to be a timing problem.  Instead, it is a problem with the Pool not having a `spec.name` (as distinct from `metadata.name`).

**Comment by jonkerj** [comment]
Indeed, and also: I appeared to be running an old version of the controller-manager (`:latest` != latest). When I change the tag to `v0.1.0-alpha.3`, it actually works.

I'll change the name of the issue.
